apps/cowswap-frontend/src/modules/notifications/hooks/useTgSubscription.ts

```diff
@@ -0,0 +1,86 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+
+import { getCmsClient } from '@cowprotocol/core'
+
+import { TgAuthorization } from './useTgAuthorization'
+
+type SubscriptionApiCaller = (method: string, data: TelegramData) => Promise<{ data: boolean }> | undefined
```

Comment:
Don't you need to import `TelegramData`?

Reply:
It's defined `react-app-env.d.ts` because it's a part of window.Telegram, so it's available globally